uberpoet/cpprojectgen.py

```diff
@@ -75,26 +79,51 @@ def example_command(self):
 
     # Generation Functions
 
-    def gen_app(self, app_node, node_list, target_loc, loc_json_file_path):
+    def gen_app(self, app_node, node_list, target_swift_loc, target_objc_loc, loc_json_file_path):
         library_node_list = [n for n in node_list if n.node_type == ModuleNode.LIBRARY]
 
         if loc_json_file_path:
             loc_reader = locreader.LocFileReader()
             loc_reader.read_loc_file(loc_json_file_path)
             loc = loc_reader.loc_for_module(n.name)
-            module_index = {n.name: {"files": self.gen_lib_module(n, loc), "loc": loc} for n in library_node_list}
+            module_index = {
+                n.name: {
+                    "files": self.gen_lib_module(n, loc, Language.SWIFT),
+                    "loc": loc,
+                    "language": Language.SWIFT
+                } for n in library_node_list
+            }
         else:
             total_code_units = 0
             for l in library_node_list:
                 total_code_units += l.code_units
 
-            loc_per_unit = target_loc / total_code_units
-            module_index = {
-                n.name: {
-                    "files": self.gen_lib_module(n, loc_per_unit),
-                    "loc": loc_per_unit
-                } for n in library_node_list
-            }
+            total_loc = target_swift_loc + target_objc_loc
+
+            swift_module_count_percentage = round(float(target_swift_loc) / total_loc, 2)
+            objc_module_count_percentage = round(float(target_objc_loc) / total_loc, 2)
+
+            split_library_node_list = list(
+                percentage_split(library_node_list, [swift_module_count_percentage, objc_module_count_percentage]))
+
+            swift_library_node_list = split_library_node_list[0]
+            objc_library_node_list = split_library_node_list[1]
+
+            loc_per_unit = total_loc / total_code_units
+            module_index = {}
+            for n in swift_library_node_list:
```

Comment:
is there a more pythonic way to accomplish looping through both lists and then adding to the module index? I think there is a zip_longest method?

Reply:
> You do have an out with list comprehensions somewhat and some libraries. You can convert these for loops into 2 separate dictionary comprehensions and then combine them with a module_index = objc_mods + swift_mods

thanks, will do that.